pyquil/api/_compiler_client.py

```diff
@@ -164,17 +172,19 @@ def __init__(self, *, client_configuration: QCSClientConfiguration, request_time
 
         self.base_url = base_url
         self.timeout = request_timeout
+        if event_loop is None:
+            event_loop = asyncio.get_event_loop()
+        self._event_loop = event_loop
 
     def get_version(self) -> str:
         """
         Get version info for compiler server.
         """
-        with self._rpcq_client() as rpcq_client:  # type: rpcq.Client
-            version: Optional[str] = rpcq_client.call("get_version_info").get("quilc")
-            if version is None:
-                raise ValueError("Expected compiler version info to contain a 'quilc' field.")
 
-            return version
+        async def _get_quilc_version() -> str:
+            return await qcs_sdk.get_quilc_version()
+
+        return self._event_loop.run_until_complete(_get_quilc_version())
```

Comment:
Is this a pattern we want to double down on? Should we create a "sync" wrapper for the qcs_sdk?

Reply:
> Should we create a "sync" wrapper for the qcs_sdk?

It would certainly make things cleaner if our bindings in other languages could just choose sync vs. async instead of having to wrap everywhere we use the SDK.

Reply:
Yeah, it would make things a lot cleaner. Looking into it, I wonder how feasible this approach is or if it can be re-used in a generic wrapper. The way we are using [get_event_loop](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop) here is deprecated, so we definitely have to consider a different approach.

Reply:
I created an issue in qcs-sdk-rust to explore this further: [qcs-sdk-rust#213](https://github.com/rigetti/qcs-sdk-rust/issues/213)